### ISSUE TITLE
[a11y]: Modal of kind danger is not accessible via keyboard

### ISSUE THREAD

**Issue by VishwaKumar** [carbon-design-system/carbon#18258]
### Package

@carbon/react

### Browser

Firefox, Chrome, Safari, Edge

### Operating System

MacOS

### Package version

v1.72.0

### React version

v17 and above

### Automated testing tool and ruleset

Test tool: Keyboard_Firefox

### Assistive technology

Test tool: Keyboard_Firefox

### Description

Not able to access modal of kind danger via keyboard from a MenuButton

**Example that shows the issue:** 
https://codesandbox.io/p/sandbox/carbondesignsystems-gvwqrt?file=%2Fsrc%2FApp.js

### WCAG 2.1 Violation

WCAG Mapping: 2.1.1 Keyboard

### Reproduction/example

https://codesandbox.io/p/sandbox/carbondesignsystems-gvwqrt?file=%2Fsrc%2FApp.js

### Steps to reproduce

In the reproduction example link,
1. Access actions menu via tab and hit enter to open
2. First menu item is selected, hit enter which will open a Edit modal ( which is of default kind )
3. Go to second menu item using down arrow, hit enter and you will see modal does not open ( which is of danger kind )

### Suggested Severity

Severity 1 = Must be fixed ASAP. The response must be swift. Someone from the team must drop all current work and be immediately reassigned to address the issue.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
It's accessible via keyboard if you select the second menu item using the spacebar instead of <kbd>enter</kbd>.
